lib/bugsnag/exception_notification.rb

```diff
@@ -12,19 +12,22 @@ def call(exception, options={}, &block)
 
       wrapped_block = proc do |report|
         options.each do |key, value|
-          # NOTE: `:env` option can be passed as a Rack Env object by `ExceptionNotification::Rack`.
-          #       Bugsnag provides the way to report the Env object by default.
-          #
-          # See below:
-          # - https://github.com/smartinez87/exception_notification/blob/2443af19e4c7433c7439c6ff01922a54023b89dd/lib/exception_notification/rack.rb#L51
-          # - https://github.com/bugsnag/bugsnag-ruby/blob/46d345d93dcb715c977615d24b369da204ed7b72/lib/bugsnag/middleware/rack_request.rb#L74-L76
-          # - https://docs.bugsnag.com/platforms/ruby/rails/configuration-options/#send_environment
-          if key == :env
-            if report.configuration.send_environment
-              report.add_tab(:environment, value)
-            end
-          else
+          case
+          when support_rack?(key)
+            process_rack(report, value)
+          when support_sidekiq?(key, value)
+            process_sidekiq(report, value)
+          when support_resque?(key, value)
+            process_resque(report, value)
+
+          # Support any attributes.
+          when report.respond_to?("#{key}=")
             report.public_send("#{key}=", value)
+
+          # Just warn instead of an error because it's tough to support
+          # all data formats which ExceptionNotification may pass.
+          else
+            warn "#{self.class.name} does not know the data: `#{key.inspect}=>#{value.inspect}`"
```

Comment:
[note] For a similar reason as [the above comment](https://github.com/pocke/exception_notification-bugsnag/pull/16/files#r255298179), I decided to just warn instead of raising an error.

But perhaps there may be better solution. 🤔